### ISSUE TITLE
Create script to install AWS gem onto a PAT worker

### ISSUE THREAD

**Issue by ckirney** [canmet-energy/btap_tasks#243]
### Description
The forthcoming gem which copies PAT results to an S3 bucket will need the AWS gem installed to work.  Need to install this gem on workers.

### Approach
Modify NREL's 'set_gem_version.sh' script to do this (while it also sets the appropriate version of openstudio-standards'.  The gem can be found here:
https://github.com/NREL/OpenStudio-server/blob/2.8.1/doc/initialization_scripts/set_gem_version.sh

### Testing Plan
Try it and see if it works.  Not sure how to test worker initialization scripts.

### Waiting On
Time.

### Repositories Involved

https://github.com/canmet-energy/btap_costing

### Select the Project Code you will be charging to 2018-2019

| Client 	|WSBE  	|Name   	|Manager   	
|----------	|---	|---	|---	
|PERD   |P-002518.001.03	|Building Archetype	|Phylroy Lopez

**Comment by ckirney** [comment]
This expanded to copying the osw files from the worker to S3.  Currently this consists of three parts.  First is the btap_set_gems.sh script.  This is a worker initialization script which replaces the standard version of openstudio-standards with our own special nrcan version (eventually in nrcan branch but right now in nrcan_280 branch because we haven't finished moving to 2.8.1).  The next part is the s3transfer_and_clean.sh script.  This installs the aws-sdk-s3 gem, downloads and runs transfer_res_s3.rb (more on that later) and then deletes most of the results so that only a minimum of information is transferred from the worker to the server.  The transfer_res_s3.rb ruby script finds the osw file on the worker and moves it to the S3 bucket in AWS.  The name it uses is the analysis_id/datapoint_id.  This means all of the datapoints for a given analysis are put in the same analysis folder on S3.  The S3 bucket I use is btapresultsbucket.

**Comment by ckirney** [comment]
This has task has been sort of completed and pulled into master.  However, on large runs only a fraction of the osw files are being transferred to S3.  I did a run with all three NECB versions, all of the archetypes (minus hospital and outpatient) and all of the cities.  This ended up with around 2500 (I think) datapoints.  I used one server and 10 workers.  The server page said everything completed successfully however only about 850 of the osw files made their way to the S3 bucket.  In very small scale runs (1 server, no workers) with only 8 datapoints all of the osw files ended up in the s3 bucket.  The command to upload the osw file to the s3 bucket will likely have to change to one that puts out more error information so that we can see what is going wrong.

**Comment by ckirney** [comment]
Have revised how osw files are transferred to s3 and how information is extracted from those files and how they are collated.  See https://github.com/canmet-energy/btap_costing/tree/master/pat_projects for more information.  Will consider this issue closed and any new modifications to the process should be made in another task.
